### PR TITLE
WIP: Make ChiantiThermalSpectrum an Astropy Model

### PR DESCRIPTION
**Pull request by DanRyanIrish**
This PR coverts the ```ChiantiThermalSpectrum``` class to an astropy model.  This is needed in order to use the astropy fitting infrastructure for spectral fitting down the road.

- [ ] Get this model working with fitting
- [ ] Update tests

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
ping @hayesla.  You can follow along my attempt to use astropy models in this PR.

**Comment by hayesla** [comment]
thanks for updating @DanRyanIrish  - seems to be working, I think the difficultly now will be getting the fitting stuff involved within it ....

but seems to plot 💃 
```
>>> from sunxspex import thermal_spectrum
>>> from astropy import units as u
>>> import numpy as np
>>> import matplotlib.pyplot as plt
>>> f_vth = thermal_spectrum.ChiantiThermalSpectrum(11*u.MK)  
>>> energy_range = np.arange(3, 50, 0.5)*u.keV
>>> temperature = 11*u.MK
>>> EM = 1e44*u.cm**-3
>>> spectrum_vals = f_vth.evaluate(energy_range, temperature, EM)
>>> plt.loglog((energy_range[1:] + energy_range[:-1])/2, spectrum_vals, drawstyle='steps-mid')
>>> plt.xlabel('Energy ({:s})'.format(energy_range.unit.to_string())
>>> plt.ylabel('Flux ({:s})'.format(spectrum_vals.unit.to_string())ß
>>> plt.show()
```

Figure_1

**Comment by DanRyanIrish** [comment]
Yay!  Look at that!  Thanks for playing with it!

**Comment by samaloney** [comment]
We might be better wrapping the class with another class or decorator that way we can use other fitting methods in the future  this way we are tightly bout to astropy modelling. 

```python
ChiantiThermalSpectrumModel(Fittable1DModel)
    def __init__(self, ....):
        model = ChiantiThermalSpectrum(...)

    def evaluate(...):
        model.evaluate(....)

ChiantiThermalSpectrum:
    ....
```

**Comment by hayesla** [comment]
Yes agree with @samaloney about this

**Comment by Cadair** [comment]
I disagree, that's horrible :stuck_out_tongue:.

You should still be able to use the model class with other fitters (the `evaluate` method is just a function at the end of the day).

**Comment by hayesla** [comment]
I guess it's a bit of a round about way! 

I haven't strayed too far away from the typical cases of `astropy.models` too much - so much stuff to discuss about the design of what we want this to look like 🙈

**Comment by samaloney** [comment]
A discussion topic so - I guess what I was really referring to was the inheritance over composition. The way `astropy.models` does model composition may not be sufficient for what we need. Also I think it could be done in a much nicer way metaclasses, decorator etc

**Comment by DanRyanIrish** [comment]
> I haven't strayed too far away from the typical cases of `astropy.models` too much - so much stuff to discuss about the design of what we want this to look like 🙈

I think this is an important point.  I think are still at a proof of concept stage where we are just trying to get stuff that works.  I suspect that we will learn a lot as part of that but that at some point not too far away we will have to stop and redesign the package organization and the API. 
 The lessons we learn from these POC exercises will be of great help.

**Comment by DanRyanIrish** [comment]
> A discussion topic so - I guess what I was really referring to was the inheritance over composition. The way `astropy.models` does model composition may not be sufficient for what we need. Also I think it could be done in a much nicer way metaclasses, decorator etc

I share your concern @samaloney that we may need some more customized model composition...maybe

File: sunxspex/thermal_spectrum.py [hunk 1]
```diff
@@ -14,22 +16,43 @@
...
-    @u.quantity_input(energy_edges=u.keV, observer_distance='length')
-    def __init__(self, energy_edges, linefile=FILE_IN,
-                 xray_abundance_file=None, abundance_type=None,
-                 observer_distance=1*u.AU, date=None):
+    # Define parameters
+    temperature = Parameter(unit=TEMPERATURE_UNIT)
+    emission_measure = Parameter(default=1e44, unit=EM_UNIT)
+
+    @u.quantity_input(temperature=TEMPERATURE_UNIT, emission_measure=EM_UNIT, observer_distance='length')
+    def __init__(self, temperature, emission_measure=emission_measure.default*emission_measure.unit,
```
[the remark this answers is not in the document]
Comment (DanRyanIrish) [reply sunxspex/thermal_spectrum.py]: Hmmm.  That's not what the online example found suggested, but perhaps I just interpretted it incorrectly.  Will test it out.

**Pull request closed without merging**